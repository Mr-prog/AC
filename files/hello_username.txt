  section_data
  question $ What is your name?
  hello $ Hello, 
  mark $ !
  section_program
  start:
    LOADMEMORY question R6
    LOADMEMORY hello R7
    LOADMEMORY mark R8
    WRITE_PORT R6
    READ_PORT
    WRITE_PORT R7
    WRITE_PORT R4
    WRITE_PORT R8
    NOP