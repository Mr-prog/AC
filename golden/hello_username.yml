in_source: |
  LOADI "What is your name?", R1
  WRITE_PORT R1
  READ_PORT R2
  LOADI "Hello, ", R3
  WRITE_PORT R3
  WRITE_PORT R2
  LOADI "!", R4
  WRITE_PORT R4
  NOP
in_stdin: "Alice"
out_code: |
  opcode=LOADI operands_type=TWO operands=['What is your name?', 1]
  opcode=WRITE_PORT operands_type=ONE operands=[1]
  opcode=READ_PORT operands_type=ONE operands=[2]
  opcode=LOADI operands_type=TWO operands=['Hello, ', 3]
  opcode=WRITE_PORT operands_type=ONE operands=[3]
  opcode=WRITE_PORT operands_type=ONE operands=[2]
  opcode=LOADI operands_type=TWO operands=['!', 4]
  opcode=WRITE_PORT operands_type=ONE operands=[4]
  opcode=NOP operands_type=NONE operands=[]
out_stdout: "What is your name?Hello, Alice!"
out_log: |
  DEBUG:translator:Start translation...
  DEBUG:machine:Running program...
